### ISSUE TITLE
Add faketime test script

### ISSUE THREAD

**Issue by yuhaoth** [Mbed-TLS/mbedtls#7729]
### Suggested enhancement

The scripts will enable faketime and run components that does not conflicts with faketime. And the script will be run in [new night build Jenkins job](https://github.com/Mbed-TLS/mbedtls-test/issues/106).

~~This script should include components in `all.sh` that does not use ASAN ( that is conflict with faketime). It should have similar interface with `all.sh` and enable faketime when running.~~

This script pick up components in `all.sh` that can cover all files in tests/data_files.

It is a nice to have feature for 2023Q2 . Related: [run this script on the CI](https://github.com/Mbed-TLS/mbedtls-test/issues/106)

### Justification

Mbed TLS needs this because we should detect which certificate are going to expire. If found in time, we can update them before they expire.

**Comment by gilles-peskine-arm** [comment]
I don't think we need to run all the components of `all.sh` for that. Do we need more than one run with `config.py full; make test; ssl-opt.sh`?

**Comment by yuhaoth** [comment]
> I don't think we need to run all the components of `all.sh` for that. Do we need more than one run with `config.py full; make test; ssl-opt.sh`?

I will double check that. 
I remember some certificates are not be covered by `config.py full`.  I do not remember which file it is.
